sql/02-bde_control_functions.sql.in

```diff
@@ -1947,56 +1947,39 @@ BEGIN
         RAISE NOTICE 'Dropping table %', v_tmptable;
         EXECUTE 'DROP TABLE ' || v_tmptable;
     ELSE
-        -- Get dependent object SQL
-
-        v_task := 'Retrieving dependent object information';
-        RAISE NOTICE '%', v_task;
-
-        v_depsql := bde_control._bde_GetDependentObjectSql(p_upload,v_bdetable);
 
         -- Is this too expensive on pg?
 
-        v_task := 'Counting current and new version of table';
+        v_task := 'Counting current version of table';
         RAISE NOTICE '%', v_task;
 
         EXECUTE 'SELECT COUNT(*) FROM ' || v_bdetable INTO v_ndel;
-        EXECUTE 'SELECT COUNT(*) FROM ' || v_tmptable INTO v_nins;
         v_nupd := 0;
 
         -- Replace the BDE table with the new version
 
-        v_task := 'Dropping current version of the table ' || v_bdetable;
+        v_task := 'Truncating current version of the table ' || v_bdetable;
         RAISE NOTICE '%', v_task;
 
-        PERFORM bde_control._bde_GetExclusiveLock(p_upload,v_bdetable);
-
         IF table_version.ver_is_table_versioned(v_bde_schema, p_table_name)
         THEN
             PERFORM table_version.ver_disable_versioning(
                         v_bde_schema, p_table_name);
         END IF;
 
-        v_sql := 'DROP TABLE ' || v_bdetable || ' CASCADE';
+        v_sql := 'TRUNCATE TABLE ' || v_bdetable;
```

Comment:
Is it safer to support only DELETE here, rather than Truncate until we only support PG > 11. See https://www.2ndquadrant.com/en/blog/upgrading-to-postgresql-11-with-logical-replication/ "PostgreSQL 11 will support replicating TRUNCATE, but that will only work if both the source and the destination instance are PostgreSQL 11 or newer." But not a big concern in the case of the production use of the BDE processor, only on the first base level full level 0 import might have issues with TRUNCATE

Reply:
Rest of the code changes look great.

Reply:
I'm ok with DELETE. We can save some time from the `COUNT(*)` which would then be replaced with diagnostics about how many rows we deleted.

Reply:
Done with next commit. Now I wonder why we're disabling revisioning on the table at all... shall we drop that part too ?

Reply:
yes we can change that part too. It will really create a mess if someone has enabled versioning, but at least the logic is better.

Reply:
Let's postpone that part, so this PR remains focused (early test show that if we don't drop the versioning we also need to create a revision...)